gtk4/src/subclass/font_chooser.rs

```diff
@@ -0,0 +1,231 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::FontChooser;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, GString, ObjectExt, Quark};
+use once_cell::sync::Lazy;
+use pango::{FontFace, FontFamily, FontMap};
+
+#[derive(Debug)]
+pub struct FilterCallback {
```

Comment:
I wonder if we should provide some way to convert a rust callback to a FilterCallback or do we expect the user to write unsafe code in order to implement the FontChooser?

Reply:
That safe function already exists: https://gtk-rs.org/gtk4-rs/gtk4/prelude/trait.FontChooserExt.html#tymethod.set_filter_func

You get this callback passed via C and in the implementation side of the font chooser you only need to pass it around and call it.

Reply:
I.e. you need to also provide a way to safely call this, see https://github.com/gtk-rs/gtk-rs/blob/c58fab4fec1254244f598c4eac7f9310184c5996/gtk/src/subclass/container.rs#L258-L266

And you'll have to understand how this works memory management wise. Does the subclass own it and has to free it? Or the base class? You will probably have to implement `Drop` on this, which then would pass the `user_data` into the `destroy_notify` (if the `destroy_notify` is not `NULL`).